plugins/modules/ndo_ipsla_track_list.py

```diff
@@ -0,0 +1,536 @@
+        if match:
+            mso.existing = match.details
+        else:
+            mso.existing = {}
+    elif module.check_mode and state != "query":
+        mso.existing = mso.proposed if state == "present" else {}
+
+    mso.exit_json()
+
+
+def get_bd_uuid(mso, schema_obj, template, bd):
+    # Get template
+    templates = [t.get("name") for t in schema_obj.get("templates")]
+    if template not in templates:
+        mso.fail_json(msg="Provided template '{0}' does not exist. Existing templates: {1}".format(template, ", ".join(templates)))
+    template_idx = templates.index(template)
+    # Get BD
+    bds = [b.get("name") for b in schema_obj.get("templates")[template_idx]["bds"]]
+    if bd not in bds:
+        mso.fail_json(msg="Provided BD '{0}' does not exist. Existing BDs: {1}".format(bd, ", ".join(bds)))
+    return schema_obj.get("templates")[template_idx]["bds"][bds.index(bd)].get("uuid")
+
+
+def get_l3out_uuid(l3out_template_object, name):
+    l3outs = l3out_template_object.template.get("l3outTemplate", {}).get("l3outs", [])
+    match = l3out_template_object.get_object_by_key_value_pairs(
+        "L3Out",
+        l3outs,
+        [KVPair("name", name)],
+        fail_module=True,
+    )
+    if match:
+        return match.details.get("uuid")
+
+
+def get_ipsla_monitoring_policy_uuid(tenant_template_obj, name):
+    existing_ipsla_policies = tenant_template_obj.template.get("tenantPolicyTemplate", {}).get("template", {}).get("ipslaMonitoringPolicies", [])
+    match = tenant_template_obj.get_object_by_key_value_pairs(
+        "IPSLA Monitoring Policy",
+        existing_ipsla_policies,
+        [KVPair("name", name)],
+        fail_module=True,
+    )
+    if match:
+        return match.details.get("uuid")
+
+
+def format_track_list_members(mso, members, obj_cache):
+    track_list_members = []
+
+    def get_obj_uuid(type, uuid, obj):
+        if uuid:
+            return uuid
+
+        name = obj.get("name")
+        template = obj.get("template")
+
+        if type == "bd":
+            schema_name = obj.get("schema")
+            if not schema_name:
+                mso.fail_json(msg="A member scope_type is bd and scope is used but the schema option is missing.")
+            schema_obj = obj_cache.get("schema-{0}-{1}".format(type, schema_name))
+            if not schema_obj:
+                id, path, schema_obj = mso.query_schema(schema_name)
+                obj_cache[schema_name] = schema_obj
+            return get_bd_uuid(mso, schema_obj, template, name)
+
+        key = "template-{0}-{1}".format(type, template)
+        mso_template = obj_cache.get(key)
+
+        if type == "l3out":
+            if not mso_template:
+                mso_template = MSOTemplate(mso, type, template)
+                mso_template.validate_template(type)
+                obj_cache[key] = mso_template
+            return get_l3out_uuid(mso_template, name)
+
+        elif type == "tenant":
+            if not mso_template:
```

Comment:
If template is not required as in if NDO doesn't allow the user to associate an IPSLA from a different template then we can directly use mso_template which was initialized the initial instantiation of MSOTemplate in line 379

Reply:
Done